tests/system-tests/rdscore/internal/rdscorecommon/egress-ip.go

```diff
@@ -0,0 +1,711 @@
+)
+
+var (
+	egressIPPodLabelsMap = map[string]string{
+		strings.Split(RDSCoreConfig.EgressIPPodLabel, "=")[0]: strings.Split(RDSCoreConfig.EgressIPPodLabel, "=")[1]}
+
+	egressIPPodSelector = metav1.ListOptions{
+		LabelSelector: RDSCoreConfig.EgressIPPodLabel,
+	}
+	nonEgressIPPodSelector = metav1.ListOptions{
+		LabelSelector: nonEgressIPPodLabel,
+	}
+)
+
+// createAgnhostRBAC creates the SCC privileged, serviceAccount and RBAC.
+func createAgnhostRBAC(
+	apiClient *clients.Settings,
+	egressIPNamespace string) (string, error) {
+	proberDeploySAName := fmt.Sprintf("rdscore-prober-sa-%s", egressIPNamespace)
+	proberDeployRBACName := fmt.Sprintf("privileged-rdscore-rbac-%s", egressIPNamespace)
+
+	var err error
+
+	glog.V(100).Infof("Adding SCC privileged to the agnhost namespace")
+
+	err = scc.AddPrivilegedSCCtoDefaultSA(egressIPNamespace)
+	if err != nil {
+		return "", fmt.Errorf("failed to add SCC privileged to the agnhost namespace %s: %w",
+			egressIPNamespace, err)
+	}
+
+	glog.V(100).Infof("Removing ServiceAccount")
+
+	err = apiobjectshelper.DeleteServiceAccount(apiClient, proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to remove serviceAccount %q from egressIPNamespace %q",
+			proberDeploySAName, egressIPNamespace)
+	}
+
+	glog.V(100).Infof("Creating ServiceAccount")
+
+	err = apiobjectshelper.CreateServiceAccount(apiClient, proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to create serviceAccount %q in egressIPNamespace %q",
+			proberDeploySAName, egressIPNamespace)
+	}
+
+	glog.V(100).Infof("Removing Cluster RBAC")
+
+	err = apiobjectshelper.DeleteClusterRBAC(apiClient, proberDeployRBACName)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to delete prober RBAC %q", proberDeployRBACName)
+	}
+
+	glog.V(100).Infof("Creating Cluster RBAC")
+
+	err = apiobjectshelper.CreateClusterRBAC(apiClient, proberDeployRBACName, proberRBACRole,
+		proberDeploySAName, egressIPNamespace)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to create prober RBAC %q in egressIPNamespace %s",
+			proberDeployRBACName, egressIPNamespace)
+	}
+
+	return proberDeploySAName, nil
+}
+
+// cleanUpDeployments removes all qe deployments leftovers from the cluster according to the label.
+func cleanUpDeployments(apiClient *clients.Settings) error {
+	namespacesList := []string{RDSCoreConfig.EgressIPNamespaceOne, RDSCoreConfig.EgressIPNamespaceTwo,
+		nonEgressIPNamespace}
+	podLabelsList := []string{RDSCoreConfig.EgressIPPodLabel, nonEgressIPPodLabel}
+
+	for _, nsName := range namespacesList {
+		deploymentsList, err := deployment.List(APIClient, nsName)
+
+		if err != nil {
+			errorRegex := fmt.Sprintf("Failed to list deployments in the namespace %s due to", nsName)
+
+			glog.V(100).Infof("Error listing deployments in the namespace %s, %v", nsName, err)
+
+			if strings.Contains(err.Error(), errorRegex) {
+				return nil
+			}
```

Comment:
I'm still trying to understand what's going on here - if there's an error listing deployments in the namespace why it's not propagated to the caller?
Code from `deployment` package doesn't return an error if there are no deployments in the requested namespace, but returns an empty slice. Empty namespace passed to the _List_ method, on the other hand, returns error